### PR TITLE
Show disabled add-on error pages

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-frontend/issues/2261

**Before:**

screenshot 2017-10-30 20 10 56

**After** 

Developer disabled add-ons:

screenshot 2017-10-30 20 09 01

Admin disabled add-ons:

screenshot 2017-10-30 20 10 06

All other 404s:

screenshot 2017-10-31 09 57 57

### CODE REVIEW

File: src/amo/components/ErrorPage/NotFound/index.js [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
       what you were looking for, and we'll do our best to fix it.`),
     { url: 'https://github.com/mozilla/addons-frontend/issues/new/' });
 
+    let explanation;
+    if (errorCode === ERROR_ADDON_DISABLED_BY_DEV) {
+      explanation =
+        i18n.gettext('This add-on has been removed by its author.');
+    } else if (errorCode === ERROR_ADDON_DISABLED_BY_ADMIN) {
+      explanation =
+        i18n.gettext('This add-on has been disabled by an administrator.');
```
Comment (tofumatt) [inline src/amo/components/ErrorPage/NotFound/index.js]: Should we just say "disabled by Mozilla"? I think that makes more sense and is a bit clearer.

File: src/amo/components/ErrorPage/NotFound/index.js [hunk 2]
```diff
@@ -1,30 +1,47 @@
...
+    } else {
+      explanation = i18n.gettext(`
+        Sorry, but we can't find anything at the address you entered.
+        If you followed a link to an add-on, it's possible that add-on
+        has been removed by its author.`);
```
Comment (tofumatt) [inline src/amo/components/ErrorPage/NotFound/index.js]: This is no longer true, right? Is there even an instance where we don't get the "removed by author" code if this is the case?

File: src/core/reducers/errors.js [hunk 3]
```diff
@@ -11,19 +17,20 @@ import log from 'core/logger';
...
   log.info('Extracting messages from error object:', error);
 
+  const logCodeChange = ({ oldCode, newCode }) => {
+    log.warn(`Replacing error code ${oldCode} with ${newCode}`);
+  };
+
   if (error && error.response && error.response.data) {
+    // Extract a code and messages from the JSON response.
     Object.keys(error.response.data).forEach((key) => {
       const val = error.response.data[key];
```
Comment (tofumatt) [inline src/core/reducers/errors.js]: I know this isn't your fault, but can we change `val` to something more meaningful? At the very least, `value` 😄

File: src/amo/components/ErrorPage/NotFound/index.js (hunk 1 above)
Comment (kumar303) [reply src/amo/components/ErrorPage/NotFound/index.js]: I got the copy from the old site. I was thinking about it and I like how administrator is a bit more AMO-focused and inclusive, meaning we have volunteers who may be helping to disable things (well, I know we have volunteer reviewers but I'm not sure about admins).

File: src/amo/components/ErrorPage/NotFound/index.js (hunk 2 above)
Comment (kumar303) [reply src/amo/components/ErrorPage/NotFound/index.js]: Oh yeah, that's true. I can shorten it.

File: src/core/reducers/errors.js (hunk 3 above)
Comment (kumar303) [reply src/core/reducers/errors.js]: Sure

**Pull request merged**